controllers/storagecluster/reconcile.go

```diff
@@ -723,3 +734,46 @@ func validateOverprovisionControlSpec(sc *ocsv1.StorageCluster, reqLogger logr.L
 	}
 	return nil
 }
+
+func validateCustomStorageClassNames(sc *ocsv1.StorageCluster) error {
+	scMap := make(map[string]string)
+	if sc.Spec.ManagedResources.CephBlockPools.StorageClassName != "" {
+		if _, has := scMap[sc.Spec.ManagedResources.CephBlockPools.StorageClassName]; has {
+			return fmt.Errorf("duplicate storage class name provided")
```

Comment:
Error message needs more details. Which StorageClass?
Also, better to validate all names before returning error. Let's say I use same name for 5 storageclasses, it should return 1 error and not 4.

Reply:
Updated